test/tools/libtesteth/ImportTest.cpp

```diff
@@ -56,6 +56,114 @@ ImportTest::ImportTest(json_spirit::mObject const& _input, json_spirit::mObject&
+        genesisObj["gasLimit"] = toCompactHexPrefixed(m_envInfo->gasLimit());
+        genesisObj["timestamp"] = toCompactHexPrefixed(m_envInfo->timestamp() - 50);
+        testObj["genesisBlockHeader"] = genesisObj;
+        testObj["pre"] = fillJsonWithState(m_statePre);
+
+        // generate expect sections for this transaction
+        if (m_testInputObject.count("expect"))
+        {
+            State s = State(0, OverlayDB(), eth::BaseState::Empty);
+            AccountMaskMap m = std::unordered_map<Address, AccountMask>();
+            StateAndMap smap{s, m};
+            vector<size_t> stateIndexesToPrint;  // not used
+            json_spirit::mArray expetSectionArray;
+
+            for (auto const& net : _networks)
+            {
+                // Calculate the block reward
+                ChainParams const chainParams{genesisInfo(net)};
+                EVMSchedule const schedule = chainParams.scheduleForBlockNumber(1);
+                u256 const blockReward = chainParams.blockReward(schedule);
+
+                TrExpectSection search{tr, smap};
+                for (auto const& exp : m_testInputObject.at("expect").get_array())
+                {
+                    TrExpectSection* search2 = &search;
+                    checkGeneralTestSectionSearch(exp.get_obj(), stateIndexesToPrint, "", search2);
+                    if (search.second.first.addresses().size() !=
+                        0)  // if match in the expect sections for this tr found
+                    {
+                        // replace expected mining reward (in state tests it is 0)
+                        json_spirit::mObject obj =
+                            fillJsonWithState(search2->second.first, search2->second.second);
+                        for (auto& adr : obj)
+                        {
+                            if (adr.first == toHexPrefixed(m_envInfo->author()))
+                            {
+                                if (adr.second.get_obj().count("balance"))
+                                {
+                                    u256 expectCoinbaseBalance =
+                                        toInt(adr.second.get_obj()["balance"]);
+                                    expectCoinbaseBalance += blockReward;
+                                    adr.second.get_obj()["balance"] =
+                                        toCompactHexPrefixed(expectCoinbaseBalance);
+                                }
+                            }
+                        }
+
+                        json_spirit::mObject expetSectionObj;
+                        expetSectionObj["network"] = test::netIdToString(net);
+                        expetSectionObj["result"] = obj;
+                        expetSectionArray.push_back(expetSectionObj);
+                        break;
+                    }
+                }  // for exp
+            }      // for net
+
+            testObj["expect"] = expetSectionArray;
```

Comment:
This function is too big, so conisider something like
`testObj["expect"] = translateExpectSection(m_testInput.at("expect"));`

Reply:
I don't want to split this function anymore. it does it's purpose - generating blockchain tests out of state tests.

Reply:
We need a third-party opinion.

Reply:
OK, I'd merge any PR that's an improvement.